### PR TITLE
Dynamic Framework project

### PR DESCRIPTION
**Pull request by Ashton-W**
I created an Xcode project that contains an iOS dynamic framework target for SWHttpTrafficRecorder. It also contains a framework unit test target, but there were no tests to migrate to this target.
The demo project was updated to link the framework.
I moved the source files into a directory to match the convention of Xcode projects.

Framework targets could be added for watchOS and tvOS, but I didn't try their compatibility.

A framework target for OSX wasn't added as the library uses some iOS SDK stuff for file writing, but with some modifications it should work and can be added.

This also means Carthage can build this project.

I have signed the CLA.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
- 
- # Unless required by applicable law or agreed to in writing, software
- # distributed under the License is distributed on an "AS IS" BASIS,
- # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- # See the License for the specific language governing permissions and
- # limitations under the License.
- #
- #******************************************************************************/
- 
 language: objective-c
```
Comment (JinlianWang) [inline .travis.yml:18]: Can we add the license portion back? I know that Capital One is too particular over this.

File: README.md [hunk 2]
```diff
@@ -1,32 +1,49 @@
...
 
-[CocoaPods](http://cocoapods.org) is the recommended way to add SWHttpTrafficRecorder to your project. Or since the library contains only two files (SWHttpTrafficRecorder.h and SWHttpTrafficRecorder.m), you can just drop them anywhere in your project and it is done. 
+##### [CocoaPods](http://cocoapods.org)
+The recommended way to add **SWHttpTrafficRecorder** to your project. Add `SWHttpTrafficRecorder` to your `Podfile`:
+
+```ruby
+pod 'SWHttpTrafficRecorder'
+```
+
+##### [Carthage](https://github.com/Carthage/Carthage)
```
Comment (JinlianWang) [inline README.md:18]: Thanks for adding the Carthage support.

File: UnitTests/SWHttpTrafficRecorderTests.m [hunk 3]
```diff
@@ -0,0 +1,39 @@
+//
+//  SWHttpTrafficRecorderTests.m
```
Comment (JinlianWang) [inline UnitTests/SWHttpTrafficRecorderTests.m:2]: We have already an "empty implementation" tests file. Do we need another one?

**Comment by JinlianWang** [comment]
Thanks for cleaning things up and add the Carthage support. It looks great overall. Will spend some time to look further to see whether the change for "Added Xcode project for framework target with unit tests" will be necessary.

File: .travis.yml (hunk 1 above)
Comment (Ashton-W) [inline .travis.yml:18]: Sure, no problem

**Comment by Ashton-W** [comment]
The framework target is required for Carthage, and for users that wish to add an Xcode project as a sub-project in their project.

**Pull request merged**